File: Recipes.Api/Application/Dto/EditRecipeDetailDto.cs

namespace Recipes.Api.Application.Dto
{
    public class EditRecipeDetailDto
    {
        public string Title { get; set; }
        public string Description { get; set; }
        public string[] Keywords { get; set; }
        public string ImagePath { get; set; }
        public int TimeInMinutes { get; set; }
        public int PersonsCount { get; set; }
        public IngredientDto[] Ingredients { get; set; }
        public StepDto[] Steps { get; set; }
    }
}

File: RecipesFrontend/src/app/dto/edit-recipe-detail/edit-recipe-detail-dto.ts
import { IngredientDto } from "../ingredient/ingredient-dto";
import { StepDto } from "../step/step-dto";

export interface EditRecipeDetailDto {
  title: string,
  description: string,
  keywords: string[],
  imagePath: string, 
  timeInMinutes: number,
  personsCount: number,
  ingredients: IngredientDto[],
  steps: StepDto[],
}
